### PR TITLE
feat: add `delete` operation

### PR DESCRIPTION
**Pull request by S-Ali-S**
This PR contains the logic for single document delete functionality. By receiving the original document and using the `delete(key)` on the store map.

### CODE REVIEW

File: src/Operations.ts [hunk 1]
```diff
@@ -72,4 +72,15 @@ export abstract class Operations<TypesMap extends Record<string, any>> {
...
     types.forEach((type) => this.collection(type).clear())
   }
+
+  delete<Type extends TypesMap[keyof TypesMap]>(document: Type) {
+    if (!isDocument(document)) {
+      throw new Error('Input document is not a valid document.')
+    }
+
+    const type = getDocumentType(document)
+    const key = getDocumentKey(document)
```
Comment (mammadataei) [inline src/Operations.ts:82]: I think the `DocumentCollection` should be responsible for resolving the `documentKey`.

File: src/Operations.ts [hunk 2]
```diff
@@ -72,4 +72,15 @@ export abstract class Operations<TypesMap extends Record<string, any>> {
   clear(...types: Array<keyof TypesMap>) {
     types.forEach((type) => this.collection(type).clear())
   }
+
+  delete<Type extends TypesMap[keyof TypesMap]>(document: Type) {
+    if (!isDocument(document)) {
+      throw new Error('Input document is not a valid document.')
```
Comment (mammadataei) [inline src/Operations.ts:78]: @S-Ali-S Can you improve the error message by adding more context to it?!

**Review by mammadataei — CHANGES_REQUESTED** [review]

File: src/Operations.ts (hunk 1 above)
Comment (S-Ali-S) [reply src/Operations.ts]: Please check for the changes applied to the code!

File: src/Operations.ts (hunk 2 above)
Comment (S-Ali-S) [reply src/Operations.ts]: Please check for the changes applied to the code!

**Pull request merged**